keps/sig-node/727-resource-metrics-endpoint/README.md

```diff
@@ -106,48 +169,227 @@ Labels are named in accordance with the [kubernetes instrumentation guidelines](
 
 <img src="https://user-images.githubusercontent.com/3262098/51704923-173bd800-1fcf-11e9-910d-3fd6606550f3.png" width="600" height="375">
 
 <img src="https://user-images.githubusercontent.com/3262098/51704880-02f7db00-1fcf-11e9-8034-c64f971a2204.png" width="600" height="375">
 
+### Test Plan
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+
+##### Unit tests
+
+- `<package>`: `<date>` - `<test coverage>`
+
+##### Integration tests
+
+- <test>: <link to test coverage>
+
+##### e2e tests
+
+Test the new endpoint with a node-e2e test similar to the current summary API test.
+Testgrid: https://k8s-testgrid.appspot.com/sig-node-kubelet#node-kubelet-features-master&include-filter-by-regex=ResourceMetricsAPI
+
+### Graduation Criteria
+
+Alpha:
+
+- [X] Implement the kubelet resource metrics endpoint as described above
+
+Beta:
+
+- [X] Modify the metrics server to consume the kubelet resource metrics endpoint 3 releases after it is added to the kubelet
+
+GA:
+
+- [X] Add [node-e2e test](https://github.com/kubernetes/kubernetes/pull/116897/files#diff-3859a7587ac4b3d1e162a2360b1fd2d3e88d4589be9b0bf19029fa7489294796R59-R70) to the node conformance tests
+
+### Upgrade / Downgrade Strategy
+
+
+
+### Version Skew Strategy
+
+This feature affects only the kubelet - in that it will expose the resource metrics for kubelet in a new endpoint, so there is no issue with version skew with other components.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [ ] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name:
+  - Components depending on the feature gate:
+- [x] Other
+  - Describe the mechanism: This feature exposes the /metrics/resource endpoint for kubelet, with all metrics annotated as STABLE 
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? No
+
+###### Does enabling the feature change any default behavior?
+
+It will expose the /metrics/resource endpoint for kubelet by default
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+/metrics/resource endpoint for kubelet will become available
+
+###### Are there any tests for feature enablement/disablement?
+
+Since there is no feature gate involved for this, there are no feature enablement/disablement test
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+
+###### What specific metrics should inform a rollback?
+
```

Comment:
Um since we are providing monitoring metrics in this feature, not sure how to answer this.
@dashpole need your help for this as well.

Reply:
got it, I included that as the response for this question. Thanks!